packages/ember/tests/routing/decoupled_basic_test.js

```diff
@@ -710,4 +749,1064 @@ class extends ApplicationTestCase {
+        return this.transitionTo('bar.baz').then(function () {
+          successCount++;
+        });
+      },
+
+      setupController() {
+        assert.ok(true, 'Should still invoke bar\'s setupController');
+      }
+    }));
+
+    this.add('route:bar.baz', Route.extend({
+      setupController() {
+        assert.ok(true, 'Should still invoke bar.baz\'s setupController');
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      this.handleURL(assert, '/foo/bar/baz');
+      assert.equal(this.applicationInstance.lookup('controller:application').get('currentPath'), 'foo.bar.baz');
+      assert.equal(successCount, 1, 'transitionTo success handler was called once');
+    });
+  }
+
+  ['@test Redirecting to the current target with a different context aborts the remainder of the routes'](assert) {
+    assert.expect(4);
+
+    this.router.map(function () {
+      this.route('home');
+      this.route('foo', function () {
+        this.route('bar', { path: 'bar/:id', resetNamespace: true }, function () {
+          this.route('baz');
+        });
+      });
+    });
+
+    let model = { id: 2 };
+
+    let count = 0;
+
+    this.add('route:bar', Route.extend({
+      afterModel() {
+        if (count++ > 10) {
+          assert.ok(false, 'infinite loop');
+        } else {
+          this.transitionTo('bar.baz', model);
+        }
+      }
+    }));
+
+    this.add('route:bar.baz', Route.extend({
+      setupController() {
+        assert.ok(true, 'Should still invoke setupController');
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      this.handleURLAborts(assert, '/foo/bar/1/baz');
+      assert.equal(this.applicationInstance.lookup('controller:application').get('currentPath'), 'foo.bar.baz');
+      assert.equal(this.applicationInstance.lookup('router:main').get('location').getURL(), '/foo/bar/2/baz');
+    });
+  }
+
+  ['@test Transitioning from a parent event does not prevent currentPath from being set'](assert) {
+    this.router.map(function () {
+      this.route('foo', function () {
+        this.route('bar', { resetNamespace: true }, function () {
+          this.route('baz');
+        });
+        this.route('qux');
+      });
+    });
+
+    this.add('route:foo', Route.extend({
+      actions: {
+        goToQux() {
+          this.transitionTo('foo.qux');
+        }
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let applicationController = this.applicationInstance.lookup('controller:application');
+      let router = this.applicationInstance.lookup('router:main');
+      this.handleURL(assert, '/foo/bar/baz');
+      assert.equal(applicationController.get('currentPath'), 'foo.bar.baz');
+      run(() => router.send('goToQux'));
+      assert.equal(applicationController.get('currentPath'), 'foo.qux');
+      assert.equal(router.get('location').getURL(), '/foo/qux');
+    });
+  }
+
+  ['@test Generated names can be customized when providing routes with dot notation'](assert) {
```

Comment:
lol what is this test title? I see that it is matches the previous test, but I have no idea where "dot notation" is used in this test.